source/dxgi/dxgi_impl_display.cpp

```diff
@@ -0,0 +1,317 @@
+	}
+
+#if (NTDDI_VERSION >= NTDDI_WIN11_GA)
+	// Prefer this API if the build environment and current OS support it, it can distinguish WCG from HDR
+	DISPLAYCONFIG_GET_ADVANCED_COLOR_INFO_2
+	get_color_info2                  = {};
+	get_color_info2.header.type      = DISPLAYCONFIG_DEVICE_INFO_GET_ADVANCED_COLOR_INFO_2;
+	get_color_info2.header.size      = sizeof     (DISPLAYCONFIG_GET_ADVANCED_COLOR_INFO_2);
+	get_color_info2.header.adapterId = _target_info.adapterId;
+	get_color_info2.header.id        = _target_info.id;
+
+	if (ERROR_SUCCESS == DisplayConfigGetDeviceInfo(&get_color_info2.header))
+	{
+		_hdr_supported = get_color_info2.highDynamicRangeSupported && !get_color_info2.advancedColorLimitedByPolicy;
+		_hdr_enabled   = get_color_info2.advancedColorActive && get_color_info2.highDynamicRangeUserEnabled && get_color_info2.activeColorMode == DISPLAYCONFIG_ADVANCED_COLOR_MODE_HDR;
+	}
+
+	else
+#endif
+	{
+		DISPLAYCONFIG_GET_ADVANCED_COLOR_INFO
+		get_color_info                  = {};
+		get_color_info.header.type      = DISPLAYCONFIG_DEVICE_INFO_GET_ADVANCED_COLOR_INFO;
+		get_color_info.header.size      = sizeof     (DISPLAYCONFIG_GET_ADVANCED_COLOR_INFO);
+		get_color_info.header.adapterId = _target_info.adapterId;
+		get_color_info.header.id        = _target_info.id;
+
+		if (ERROR_SUCCESS == DisplayConfigGetDeviceInfo(&get_color_info.header))
+		{
+			_hdr_supported = get_color_info.advancedColorSupported;
+			_hdr_enabled   = get_color_info.advancedColorEnabled;
+		}
+	}
+};
+
+const wchar_t* reshade::dxgi::display_impl::get_device_name() const
+{
+	return _desc.DeviceName;
+};
+
+const wchar_t* reshade::dxgi::display_impl::get_device_path() const
+{
+	return _path.c_str();
+};
+
+const wchar_t* reshade::dxgi::display_impl::get_display_name() const
+{
+	return _name.c_str();
+};
+
+reshade::api::display::monitor reshade::dxgi::display_impl::get_monitor() const
+{
+	return _desc.Monitor;
+};
+
+reshade::api::rect reshade::dxgi::display_impl::get_desktop_coords() const
+{
+	return { _desc.DesktopCoordinates.left,  _desc.DesktopCoordinates.top,
+			 _desc.DesktopCoordinates.right, _desc.DesktopCoordinates.bottom };
+};
+
+uint32_t reshade::dxgi::display_impl::get_color_depth() const
+{
+	return _desc.BitsPerColor;
+};
+
+reshade::api::rational reshade::dxgi::display_impl::get_refresh_rate() const
+{
+	return _refresh_rate;
+};
+
+reshade::api::color_space reshade::dxgi::display_impl::get_color_space() const
+{
+	switch (_desc.ColorSpace)
+	{
+	case DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709:
+		return api::color_space::srgb_nonlinear;
+	case DXGI_COLOR_SPACE_RGB_FULL_G10_NONE_P709:
+		return api::color_space::extended_srgb_linear;
+	case DXGI_COLOR_SPACE_RGB_FULL_G2084_NONE_P2020:
+		return api::color_space::hdr10_st2084;
+	case DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P2020:
+		return api::color_space::hdr10_hlg;
+	default:
+		return api::color_space::unknown;
+	}
```

Comment:
You could include `d3d12/d3d12_impl_type_convert.hpp` and just call `return d3d12::convert_color_space(_desc.ColorSpace);` here to avoid some code duplication.